packages/debug/src/browser/editor/debug-breakpoint-widget.tsx

```diff
@@ -94,47 +94,63 @@ export class DebugBreakpointWidget implements Disposable {
         inputNode.classList.add('theia-debug-breakpoint-input');
         this.zone.containerNode.appendChild(inputNode);
 
-        const input = this._input = await this.createInput(inputNode);
+        const breakpointInput = this._input = await this.createInput(inputNode);
+        const breakpointInputEditor = breakpointInput.getControl();
         if (this.toDispose.disposed) {
-            input.dispose();
+            breakpointInput.dispose();
             return;
         }
-        this.toDispose.push(input);
-        this.toDispose.push(monaco.modes.CompletionProviderRegistry.register({ scheme: input.uri.scheme }, {
+        this.toDispose.push(breakpointInput);
+        this.toDispose.push(monaco.modes.CompletionProviderRegistry.register({ scheme: breakpointInput.uri.scheme }, {
             provideCompletionItems: async (model, position, context, token) => {
                 const suggestions = [];
-                if ((this.context === 'condition' || this.context === 'logMessage')
-                    && input.uri.toString() === model.uri.toString()) {
-                    const editor = this.editor.getControl();
-                    const items = await monaco.suggest.provideSuggestionItems(
-                        editor.getModel()!,
-                        new monaco.Position(editor.getPosition()!.lineNumber, 1),
-                        new monaco.suggest.CompletionOptions(undefined, new Set<monaco.languages.CompletionItemKind>().add(monaco.languages.CompletionItemKind.Snippet)),
-                        context, token);
+                const shouldProvideAutoCompleteInContext = (this.context === 'condition') ||
+                    (this.context === 'logMessage' && DebugBreakpointWidget.isCurlyBracketOpen(breakpointInputEditor));
+                const doesInputEditorMatchModel = breakpointInput.uri.toString() === model.uri.toString();
+                if (shouldProvideAutoCompleteInContext && doesInputEditorMatchModel) {
+                    const sourceCodeEditor = this.editor.getControl();
+                    const sourceCodeEditorModel = sourceCodeEditor.getModel();
+                    const sourceCodeEditorPosition = sourceCodeEditor.getPosition();
+                    const completionItemModel = (sourceCodeEditorModel && sourceCodeEditorPosition) && await monaco.suggest.provideSuggestionItems(
+                        sourceCodeEditorModel,
+                        new monaco.Position(sourceCodeEditorPosition.lineNumber, 1),
+                        new monaco.suggest.CompletionOptions(
+                            undefined,
+                            new Set<monaco.languages.CompletionItemKind>([monaco.languages.CompletionItemKind.Snippet]),
+                        ),
+                        context,
+                        token,
+                    );
```

Comment:
It would clean things up a little to move this to its own method. Something like

```typescript
createCompletionItemModel(context, token) {
    const sourceCodeEditor = this.editor.getControl();
    const sourceCodeEditorModel = sourceCodeEditor.getModel();
    const sourceCodeEditorPosition = sourceCodeEditor.getPosition();
    return (sourceCodeEditorModel && sourceCodeEditorPosition) && await monaco.suggest.provideSuggestionItems(
        sourceCodeEditorModel,
        new monaco.Position(sourceCodeEditorPosition.lineNumber, 1),
        new monaco.suggest.CompletionOptions(
            undefined,
            new Set<monaco.languages.CompletionItemKind>([monaco.languages.CompletionItemKind.Snippet]),
        ),
        context,
        token,
    );
}
```

Reply:
Good idea, I've added this change